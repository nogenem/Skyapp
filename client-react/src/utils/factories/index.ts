import * as models from './models';
import * as states from './states';

export default {
  models,
  states,
};
