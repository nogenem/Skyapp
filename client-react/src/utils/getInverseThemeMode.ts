import { TThemeMode } from '~/redux/theme/types';

// PS: This should be done automatically by the lib... ;/
export default (mode: TThemeMode) => (mode === 'dark' ? 'light' : 'dark');
