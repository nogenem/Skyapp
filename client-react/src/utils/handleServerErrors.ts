interface IErrors {
  [x: string]: string;
}

interface IServerErrorsObject {
  [x: string]: string | string[];
}

interface IServerError {
  errors: IServerErrorsObject;
}

interface IPartialAxiosError {
  response?: {
    data: IServerError;
  };
}

// TODO: Think a way to translate the errors coming from the server
export default (err: IPartialAxiosError): IErrors => {
  if (
    !err ||
    !err.response ||
    !err.response.data ||
    !err.response.data.errors
  ) {
    if (process.env.NODE_ENV !== 'test') console.error(err);
    return { global: 'Internal server error' };
  }

  const { errors } = err.response.data;
  const ret = {} as IErrors;

  Object.keys(errors).forEach((key: string) => {
    const error: string = (
      Array.isArray(errors[key]) ? errors[key][0] : errors[key]
    ) as string;
    ret[key] = error;
  });
  return ret;
};

export type { IErrors, IServerErrorsObject, IServerError, IPartialAxiosError };
