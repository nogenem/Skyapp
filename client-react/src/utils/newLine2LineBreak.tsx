import React from 'react';

// https://stackoverflow.com/questions/40418024/how-to-replace-n-to-linebreaks-in-react-js?rq=1
export default (string: string) =>
  string
    .split('\n')
    .map((item, index) => (index === 0 ? item : [<br key={index} />, item]));
