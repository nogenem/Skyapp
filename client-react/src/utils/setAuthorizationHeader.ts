import { axiosInstance } from '~/services/ApiService';

export default (token?: string) => {
  if (token) {
    axiosInstance.defaults.headers.common.authorization = `Bearer ${token}`;
  } else {
    delete axiosInstance.defaults.headers.common.authorization;
  }
};
