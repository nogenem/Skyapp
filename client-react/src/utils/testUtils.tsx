import React from 'react';
import { Provider } from 'react-redux';

import { render } from '@testing-library/react';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import type { IAppState } from '~/redux/store';

export const getMockStore = () => {
  const middlewares = [thunk];
  return configureStore(middlewares);
};

export const getRenderWithRedux = () => {
  const emptyState: Partial<IAppState> = {};
  const mockStore = getMockStore();

  return (
    ui: React.ReactNode,
    initialState: Partial<IAppState> = emptyState,
  ) => {
    const store = mockStore(initialState);
    return {
      ...render(<Provider store={store}>{ui}</Provider>),
    };
  };
};
