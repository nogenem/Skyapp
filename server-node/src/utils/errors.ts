import {
  INTERNAL_SERVER_ERROR,
  INVALID_ID,
  ROUTE_NOT_FOUND_ERROR,
  INVALID_CREDENTIALS,
  INVALID_EXPIRED_TOKEN,
  NO_USER_WITH_SUCH_EMAIL,
  LAST_EMAIL_SENT_IS_STILL_VALID,
  NO_TOKEN,
  CHANNEL_ALREADY_EXISTS,
  USER_IS_NOT_GROUP_ADM,
  NOT_MEMBER_OF_GROUP,
  CANT_EDIT_THIS_MESSAGE,
} from '~/constants/error_messages';
import type { TTranslatableError } from '~/constants/error_messages';

interface IMsgObj<T> {
  [x: string]: T;
}
type TMsgObjTypes =
  | (string | TTranslatableError)[]
  | string
  | TTranslatableError;

export type { IMsgObj, TMsgObjTypes };

export class CustomError extends Error {
  msgObj: IMsgObj<TMsgObjTypes>;

  kind: string;

  status: number;

  constructor(msgObj: IMsgObj<TMsgObjTypes>, status = 400) {
    super(JSON.stringify(msgObj, null, 2));

    if (Error.captureStackTrace) {
      Error.captureStackTrace(this, CustomError);
    }

    this.msgObj = msgObj;
    this.kind = 'CustomError';
    this.status = status;
  }
}

export const validatorErrors = (errs: IMsgObj<TMsgObjTypes>): CustomError =>
  new CustomError(errs);

export const internalServerError = (): CustomError =>
  new CustomError({ global: INTERNAL_SERVER_ERROR }, 500);

export const invalidIdError = (): CustomError =>
  new CustomError({ global: INVALID_ID });

export const mongooseErrors = (errs: IMsgObj<string>): CustomError =>
  new CustomError(errs);

export const routeNotFoundError = (): CustomError =>
  new CustomError({ global: ROUTE_NOT_FOUND_ERROR }, 404);

export const invalidCredentialsError = (): CustomError =>
  new CustomError({ global: INVALID_CREDENTIALS });

export const invalidOrExpiredTokenError = (): CustomError =>
  new CustomError({ global: INVALID_EXPIRED_TOKEN });

export const noUserWithSuchEmailError = (): CustomError =>
  new CustomError({ email: NO_USER_WITH_SUCH_EMAIL });

export const lastEmailSentIsStillValidError = (): CustomError =>
  new CustomError({ email: LAST_EMAIL_SENT_IS_STILL_VALID });

export const noTokenError = (): CustomError =>
  new CustomError({ global: NO_TOKEN });

export const channelAlreadyExistsError = (): CustomError =>
  new CustomError({ global: CHANNEL_ALREADY_EXISTS });

export const userIsNotGroupAdmError = (): CustomError =>
  new CustomError({ global: USER_IS_NOT_GROUP_ADM });

export const notMemberOfGroupError = (): CustomError =>
  new CustomError({ global: NOT_MEMBER_OF_GROUP });

export const multerErrors = (
  errs: [msgObj: IMsgObj<TMsgObjTypes>, status?: number],
): CustomError => new CustomError(...errs);

export const cantEditThisMessageError = (): CustomError =>
  new CustomError({ global: CANT_EDIT_THIS_MESSAGE });
