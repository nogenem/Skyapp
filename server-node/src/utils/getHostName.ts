interface IHeader {
  [x: string]: string | string[] | undefined;
}

export default (headers: IHeader): string => {
  if (headers['x-forwarded-host'])
    return `${headers['x-forwarded-proto']}://${headers['x-forwarded-host']}`;

  if (headers.referer) {
    const host = headers.referer as string;
    const url = new URL(host);
    return url.origin;
  }

  let { host } = headers;
  host = (host || '') as string;
  if (!host.startsWith('http')) host = `http://${host}`;
  return host;
};
