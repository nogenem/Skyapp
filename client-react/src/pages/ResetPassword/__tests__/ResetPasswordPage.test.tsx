import React from 'react';

import { getRenderWithRedux } from '~/utils/testUtils';

import { ResetPassword as ResetPasswordPage } from '../index';

const renderWithRedux = getRenderWithRedux();

// TODO: Add better tests for this !?
// The ./Form is already being tested by this
// and the ~/components/Form is being tested there, so...
describe('ResetPasswordPage', () => {
  it('renders correctly', () => {
    const { container } = renderWithRedux(<ResetPasswordPage />);

    expect(container).toMatchSnapshot();
  });
});
