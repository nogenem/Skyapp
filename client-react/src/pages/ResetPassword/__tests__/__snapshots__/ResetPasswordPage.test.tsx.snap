// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResetPasswordPage renders correctly 1`] = `
<div>
  <div
    class="MuiGrid-root makeStyles-root-1 MuiGrid-container"
  >
    <div
      class="MuiGrid-root makeStyles-container-2 MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        class="makeStyles-paper-3"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-4 MuiAvatar-colorDefault"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM9 6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9V6zm9 14H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
            />
          </svg>
        </div>
        <h1
          class="MuiTypography-root MuiTypography-h5"
        >
          Reset Password
        </h1>
        <form
          id="reset-password-form"
          style="width: 100%;"
        >
          <div
            class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
              data-shrink="false"
              for="reset-password-form-newPassword"
              id="reset-password-form-newPassword-label"
            >
              New Password
              <span
                aria-hidden="true"
                class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
              >
                 
                *
              </span>
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
            >
              <input
                aria-invalid="false"
                autocomplete="newPassword"
                class="MuiInputBase-input MuiOutlinedInput-input"
                id="reset-password-form-newPassword"
                name="newPassword"
                required=""
                type="password"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-6 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-8"
                >
                  <span>
                    New Password
                     *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
              data-shrink="false"
              for="reset-password-form-newPasswordConfirmation"
              id="reset-password-form-newPasswordConfirmation-label"
            >
              New Password Confirmation
              <span
                aria-hidden="true"
                class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
              >
                 
                *
              </span>
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
            >
              <input
                aria-invalid="false"
                autocomplete="newPasswordConfirmation"
                class="MuiInputBase-input MuiOutlinedInput-input"
                id="reset-password-form-newPasswordConfirmation"
                name="newPasswordConfirmation"
                required=""
                type="password"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-6 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-8"
                >
                  <span>
                    New Password Confirmation
                     *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-submit-5 MuiButton-containedPrimary MuiButton-fullWidth"
            tabindex="0"
            type="submit"
          >
            <span
              class="MuiButton-label"
            >
              Reset Password
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </form>
        <div
          class="MuiGrid-root MuiGrid-container"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
          >
            <a
              class="MuiTypography-root MuiLink-root MuiLink-underlineAlways MuiTypography-body2 MuiTypography-colorTextPrimary"
              href="/signup"
            >
              Sign Up
            </a>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <a
              class="MuiTypography-root MuiLink-root MuiLink-underlineAlways MuiTypography-body2 MuiTypography-colorTextPrimary"
              href="/signin"
            >
              Sign In
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
